libraries/eosiolib/core/eosio/crypto_ext.hpp

```diff
@@ -48,14 +49,183 @@ namespace eosio {
+      };
+
+      /**
+       * Construct a point given a serialized point
+       *
+       * @param p - The serialized point
+       */
+      point_view(std::vector<char>& p)
+      :x(p.data()), y(p.data() + p.size()/2), size(p.size()/2)
+      {
+      };
+
+      /**
+       *  Return serialzed point containing only x and y
+       */
+      std::vector<char> serialized() const {
+         std::vector<char> x_and_y( x, x + size );
+         x_and_y.insert( x_and_y.end(), y, y + size );
+         return x_and_y;
+      }
+
+      /**
+      * Copy a serialized buffer to current point
+      *
+      * @param source - The source to be copied from
+      */
+      void copy_from(const std::vector<char>& source) {
+         eosio::check( source.size() == 2 * size, "size of souce buffer must be size of x + y" );
+         std::memcpy(x, source.data(), size);
+         std::memcpy(y, source.data() + size, size);
+      }
+   };
+
+   static constexpr size_t g1_coordinate_size = 32;
+   static constexpr size_t g2_coordinate_size = 64;
+   static constexpr size_t blake2f_result_size = 64;
+
+   /**
+    * Abstracts G1 point. Ensures sizes of x and y are 32
+    *
+    *  @ingroup crypto
+    */
+   struct g1_view : public point_view {
+      /**
+       * Construct a G1 point given x and y
+       *
+       * @param x_ - The x coordinate, a vector of chars
+       * @param y_ - The y coordinate, a vector of chars
+       */
+      g1_view(std::vector<char>& x_, std::vector<char>& y_)
+      : point_view(x_, y_)
+      {
+         eosio::check( size == g1_coordinate_size, "G1 coordinate size must be 32" );
+      }
+
+      /**
+       * Construct a point given a serialized point
+       *
+       * @param p - The serialized point
+       */
+      g1_view(std::vector<char>& p)
+      : point_view(p)
+      {
+         eosio::check( p.size() == g1_coordinate_size * 2, "G1 serialized size must be 64" );
+      };
+   };
+
+   /**
+    * Abstracts G2 point. Ensures sizes of x and y are 64
+    *
+    *  @ingroup crypto
+    */
+   struct g2_view : public point_view {
+      /**
+       * Construct a G2 point given x and y
+       *
+       * @param x_ - The x coordinate, a vector of chars
+       * @param y_ - The y coordinate, a vector of chars
+       */
+      g2_view(std::vector<char>& x_, std::vector<char>& y_)
+      : point_view(x_, y_)
+      {
+         eosio::check( size == g2_coordinate_size, "G2 coordinate size must be 64" );
+      }
+
+      /**
+       * Construct a point given a serialized point
+       *
+       * @param p - The serialized point
+       */
+      g2_view(std::vector<char>& p)
+      : point_view(p)
+      {
+         eosio::check( p.size() == g2_coordinate_size * 2, "G2 serialized size must be 128" );
+      };
+   };
+
+   /**
+    * Abstracts big integer.
+    *
+    *  @ingroup crypto
+    */
+   struct bigint {
+      /**
+       * Construct a bigint given a vector of chars (bytes)
+       *
```

Comment:
Given our uses of bigint at this point, I would suggest we just go with an std::vector<char> or a typedef.

So,

```c++
using bigint = std::vector<char>;
```

Reply:
I think `std::vector<char>` would be better aliased as `bytes` since it could be used in other non-crypto related cases. Also, since ABI already defines `bytes` type for `std::vector<char>` I believe such alias would also improve code readability.

Reply:
Did you mean
```
using bytes = std::vector<char>;
using bigint = bytes;
```
?

Reply:
I guess you could abstract it that way. Tho in this case, the `bigint` alias just brings another abstraction layer with no additional functionality and I don't know if such abstraction is ok in terms of code readability and usage (i.e. 2 layers `bigint` -> `bytes` to know it's an alias for `std::vector<char>`). If it were up to me I'd avoid `bigint` alias.

Reply:
Agreed. Two layers of abstraction is not good. Keep `bigint` as is for future extensions.